scripts/ci-e2e.sh

```diff
@@ -47,10 +47,12 @@ export TEST_NAME=${CLUSTER_NAME:-"capg-${RANDOM}"}
 export GCP_NETWORK_NAME=${GCP_NETWORK_NAME:-"${TEST_NAME}-mynetwork"}
 GCP_B64ENCODED_CREDENTIALS=$(base64 "$GOOGLE_APPLICATION_CREDENTIALS" | tr -d '\n')
 export GCP_B64ENCODED_CREDENTIALS
-export KUBERNETES_MAJOR_VERSION="1"
-export KUBERNETES_MINOR_VERSION="27"
-export KUBERNETES_PATCH_VERSION="3"
-export KUBERNETES_VERSION="v${KUBERNETES_MAJOR_VERSION}.${KUBERNETES_MINOR_VERSION}.${KUBERNETES_PATCH_VERSION}"
+KUBERNETES_VERSION=$(go run github.com/mikefarah/yq/v4@v4.45.4 '.variables.KUBERNETES_VERSION' test/e2e/config/gcp-ci.yaml)
```

Comment:
to be honest i dont like this 🙈

Reply:
This came out of: https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/1476#discussion_r2103197094

I am happy with either, so up to you and @justinsb to decide what you like best :)

Reply:
what is the problem with hardcoded ones? seems overkill add a go run to just parse 3 tiny items